darwin/future/core/types/query.py

```diff
@@ -2,7 +2,7 @@
 
```

Comment:
Just a generally well factored file eh

Reply:
I know, so much refactoring. At least it doesn't require 2 generics now though which makes everything easier to create.